vllm/worker/hpu_enc_dec_model_runner.py

```diff
@@ -0,0 +1,604 @@
+                                                            dtype=torch.long,
+                                                            device=self.device)
+        else:
+            for seq_group_metadata in seq_group_metadata_list:
+                for _ in range(len(seq_group_metadata.seq_data)):
+                    encoder_seq_len = (
+                        seq_group_metadata.encoder_seq_data.get_len()
+                        if seq_group_metadata.encoder_seq_data else 0)
+                    encoder_seq_lens.append(encoder_seq_len)
+                    cross_block_table = seq_group_metadata.cross_block_table
+                    cross_block_tables.append([] if (
+                        cross_block_table is None) else cross_block_table)
+
+            last_block_usage = [(encoder_seq_len - 1) % self.block_size + 1
+                                for encoder_seq_len in encoder_seq_lens]
+            block_groups = [[i] * len(bt)
+                            for i, bt in enumerate(cross_block_tables)]
+            block_usage = [
+                [self.block_size] * (len(bt) - 1) + [lbu]
+                for bt, lbu in zip(cross_block_tables, last_block_usage) if bt
+            ]
+
+            block_list = self._flatten(cross_block_tables)
+            block_groups = self._flatten(block_groups)
+            block_usage = self._flatten(block_usage)
+
+            assert len(block_list) == len(block_groups)
+            assert len(block_list) == len(block_usage)
+
+            block_list = torch.tensor(block_list,
+                                      dtype=torch.int,
+                                      device='cpu')
+            block_groups = torch.tensor(block_groups,
+                                        dtype=torch.int,
+                                        device='cpu')
+            block_usage = torch.tensor(block_usage,
+                                       dtype=self.model_config.dtype,
+                                       device='cpu')
+
+            block_list = block_list.to(  # type: ignore
+                self.device, non_blocking=True)
+            block_groups = block_groups.to(  # type: ignore
+                self.device, non_blocking=True)
+            block_usage = block_usage.to(  # type: ignore
+                self.device, non_blocking=True)
+
+            attn_metadata.cross_block_list = block_list
+            attn_metadata.cross_block_groups = block_groups
+            attn_metadata.cross_block_usage = block_usage
+
+        encoder_seq_lens_tensor = self._list_to_int32_tensor(encoder_seq_lens)
+        attn_metadata.encoder_seq_lens = encoder_seq_lens
+        attn_metadata.encoder_seq_lens_tensor = encoder_seq_lens_tensor
+
+        return attn_metadata
+
+    def profile_run(self) -> None:
+        num_layers = self.model_config.get_num_layers(self.parallel_config)
+        kv_caches = [None] * num_layers
+        max_batch_size = self.scheduler_config.max_num_seqs
```

Comment:
Using
```
max_batch_size = min(
    self.max_num_prefill_seqs,
    self.max_num_batched_tokens // max_seq_len
)
```
will also cause OOM. Because the scheduler will schedule `max_num_prefill_seqs` number of sequences. If this exceeds the `self.max_num_batched_tokens // max_seq_len`, will cause OOM.
Indeed, the `profile_run ` should run with max memory consumed. Because the remaining space will be used up by the KV cache, graph building. If the batchsize of subsequent running (such as warm-up buckets or the final running) is greater than the `profile_run`, and the reserved 10% free space didn't accommodate increased memory usage, it will OOM. I did encounter this issue.
I suggest here keep `self.max_num_prefill_seqs` unchanged to avoid OOM risks.

Reply:
It typically happens in encoder-decoder-vision models because the number of encoder tokens is much larger than decoder tokens, often around 6403. Each sequence has its own encoder sequence. So the batch size primarily dictates the memory consumption in such cases.